postgresql  ERROR while scanning media folder

Hi,

wile scanning media folder, Libresonic throws an error and stops the scan:

[2017-04-06 12:20:12,027] INFO MediaScannerService - Scanned media library with 10750 entries.
[2017-04-06 12:20:13,776] ERROR MediaScannerService - Failed to scan media library.
org.springframework.dao.DataIntegrityViolationException: PreparedStatementCallback; SQL [update media_file set folder=?,type=?,format=?,title=?,album=?,artist=?,album_artist=?,disc_number=?,track_number=?,year=?,genre=?,bit_rate=?,variable_bit_rate=?,duration_seconds=?,file_size=?,width=?,height=?,cover_art_path=?,parent_path=?,play_count=?,last_played=?,comment=?,changed=?,last_scanned=?,children_last_updated=?,present=?, version=? where path=?]; ERROR: invalid byte sequence for encoding "UTF8": 0x00; nested exception is org.postgresql.util.PSQLException: ERROR: invalid byte sequence for encoding "UTF8": 0x00
        at org.springframework.jdbc.support.SQLStateSQLExceptionTranslator.doTranslate(SQLStateSQLExceptionTranslator.java:102)
        at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:73)
        at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:81)
        at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:81)
        at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:649)
        at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:870)
        at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:931)
        at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:941)
        at org.libresonic.player.dao.AbstractDao.update(AbstractDao.java:82)
        at org.libresonic.player.dao.MediaFileDao.createOrUpdateMediaFile(MediaFileDao.java:164)
        at org.libresonic.player.dao.MediaFileDao$$FastClassBySpringCGLIB$$83a0b56b.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:204)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:720)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
        at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:99)
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:282)
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:655)
        at org.libresonic.player.dao.MediaFileDao$$EnhancerBySpringCGLIB$$17d6a963.createOrUpdateMediaFile(<generated>)
        at org.libresonic.player.service.MediaFileService.getMediaFile(MediaFileService.java:105)
        at org.libresonic.player.service.MediaFileService.getMediaFile(MediaFileService.java:67)
        at org.libresonic.player.service.MediaFileService.findCoverArt(MediaFileService.java:607)
        at org.libresonic.player.service.MediaFileService.createMediaFile(MediaFileService.java:521)
        at org.libresonic.player.service.MediaFileService.updateChildren(MediaFileService.java:393)
        at org.libresonic.player.service.MediaFileService.getChildrenOf(MediaFileService.java:179)
        at org.libresonic.player.service.MediaScannerService.scanFile(MediaScannerService.java:230)
        at org.libresonic.player.service.MediaScannerService.scanFile(MediaScannerService.java:234)
        at org.libresonic.player.service.MediaScannerService.doScanLibrary(MediaScannerService.java:172)
        at org.libresonic.player.service.MediaScannerService.access$000(MediaScannerService.java:38)
        at org.libresonic.player.service.MediaScannerService$2.run(MediaScannerService.java:142)
Caused by: org.postgresql.util.PSQLException: ERROR: invalid byte sequence for encoding "UTF8": 0x00
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2412)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2125)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:297)
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:428)
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:354)
        at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:169)
        at org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:136)
        at org.apache.tomcat.dbcp.dbcp2.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:97)
        at org.apache.tomcat.dbcp.dbcp2.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:97)
        at org.springframework.jdbc.core.JdbcTemplate$2.doInPreparedStatement(JdbcTemplate.java:877)
        at org.springframework.jdbc.core.JdbcTemplate$2.doInPreparedStatement(JdbcTemplate.java:870)
        at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:633)
        ... 26 more


I'm Using:
Version	6.2.beta1 – March 5, 2017
Server	Apache Tomcat/8.0.41, java 1.8.0_121, Linux (448.3 MB / 940.5 MB)
and postgre DB in "jndi" mode :

        driverClassName="org.postgresql.Driver"
        url="jdbc:postgresql://localhost/libresonic?stringtype=unspecified"/>

--- a/libresonic-main/pom.xml
+++ b/libresonic-main/pom.xml
@@ -194,9 +194,9 @@
         </dependency>
 
         <dependency>
-            <groupId>org</groupId>
+            <groupId>net.jthink</groupId>
             <artifactId>jaudiotagger</artifactId>
-            <version>2.0.3</version>
+            <version>2.2.3</version>
         </dependency>
 
         <dependency>
@@ -397,6 +397,17 @@
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
             <version>1.7.24</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-core</artifactId>
+            <version>2.8.7</version>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-databind</artifactId>
+            <version>2.8.7</version>
         </dependency>
     </dependencies>
 

--- a/libresonic-main/src/main/java/org/libresonic/player/dao/MediaFileDao.java
+++ b/libresonic-main/src/main/java/org/libresonic/player/dao/MediaFileDao.java
@@ -24,6 +24,7 @@ import org.libresonic.player.domain.Genre;
 import org.libresonic.player.domain.MediaFile;
 import org.libresonic.player.domain.MusicFolder;
 import org.libresonic.player.domain.RandomSearchCriteria;
+import org.libresonic.player.util.Util;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.jdbc.core.RowMapper;
@@ -164,6 +165,8 @@ public class MediaFileDao extends AbstractDao {
                      "present=?, " +
                      "version=? " +
                      "where path=?";
+
+        logger.trace("Updating media file {}", Util.debugObject(file));
 
         int n = update(sql,
                        file.getFolder(), file.getMediaType().name(), file.getFormat(), file.getTitle(), file.getAlbumName(), file.getArtist(),

--- a/libresonic-main/src/main/java/org/libresonic/player/service/metadata/JaudiotaggerParser.java
+++ b/libresonic-main/src/main/java/org/libresonic/player/service/metadata/JaudiotaggerParser.java
@@ -26,7 +26,7 @@ import org.jaudiotagger.audio.AudioFileIO;
 import org.jaudiotagger.audio.AudioHeader;
 import org.jaudiotagger.tag.FieldKey;
 import org.jaudiotagger.tag.Tag;
-import org.jaudiotagger.tag.datatype.Artwork;
+import org.jaudiotagger.tag.images.Artwork;
 import org.jaudiotagger.tag.reference.GenreTypes;
 import org.libresonic.player.Logger;
 import org.libresonic.player.domain.MediaFile;

--- a/libresonic-main/src/main/java/org/libresonic/player/util/Util.java
+++ b/libresonic-main/src/main/java/org/libresonic/player/util/Util.java
@@ -19,12 +19,16 @@
  */
 package org.libresonic.player.util;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.libresonic.player.Logger;
 
 import javax.servlet.ServletResponse;
 import javax.servlet.http.HttpServletResponse;
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 
 /**
  * Miscellaneous general utility methods.
@@ -101,5 +105,15 @@ public final class Util {
             result[i] = values.get(i);
         }
         return result;
+    }
+
+    static ObjectMapper objectMapper = new ObjectMapper();
+    public static String debugObject(Object object) {
+        try {
+            return objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(object);
+        } catch (JsonProcessingException e) {
+            LOG.warn("Cant output debug object", e);
+            return "";
+        }
     }
 }

--- a/pom.xml
+++ b/pom.xml
@@ -35,6 +35,10 @@
             <name>teleal</name>
             <url>http://teleal.org/m2</url>
         </repository>
+        <repository>
+            <id>jaudiotagger-repository</id>
+            <url>https://dl.bintray.com/ijabz/maven</url>
+        </repository>
     </repositories>
 
     <pluginRepositories>

